x-pack/plugins/fleet/server/routes/policy_preconfig/index.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IRouter, RequestHandler } from 'src/core/server';
+import type { TypeOf } from '@kbn/config-schema';
+
+import type { FleetConfigType } from '../../../common';
+
+import { PLUGIN_ID, POLICY_PRECONFIG_API_ROUTES } from '../../constants';
+import { PutPolicyPreconfigSchema } from '../../types';
+import { defaultIngestErrorHandler } from '../../errors';
+import { ensurePreconfiguredPackagesAndPolicies, outputService } from '../../services';
+
+export const putPolicyPreconfigHandler: RequestHandler<
+  undefined,
+  undefined,
+  TypeOf<typeof PutPolicyPreconfigSchema.body>
+> = async (context, request, response) => {
+  const soClient = context.core.savedObjects.client;
+  const esClient = context.core.elasticsearch.client.asCurrentUser;
+  const defaultOutput = await outputService.ensureDefaultOutput(soClient);
+
+  const { agentPolicies, packages } = request.body;
+
+  try {
+    const body = await ensurePreconfiguredPackagesAndPolicies(
+      soClient,
+      esClient,
+      (agentPolicies as FleetConfigType['agentPolicies']) ?? [],
```

Comment:
this will need to change as I requested for the changes to `FleetConfigType` to be removed 😀 perhaps the changes there can be moved to `common/types/models/agent_policy.ts` as a new type, something like
```ts
export interface PreconfiguredAgentPolicy extends NewAgentPolicy {
  ...
}
```